scripts/Makefile.host

```diff
@@ -128,5 +144,17 @@ quiet_cmd_host-cxxobjs	= HOSTCXX $@
 $(host-cxxobjs): $(obj)/%.o: $(src)/%.cc FORCE
 	$(call if_changed_dep,host-cxxobjs)
 
+# Create executable from a single Rust crate (which may consist of
+# one or more `.rs` files)
+# host-rust -> Executable
+quiet_cmd_host-rust	= HOSTRUSTC $@
+      cmd_host-rust	= \
+	$(HOSTRUSTC) $(hostrust_flags) --emit=dep-info,link \
+		--out-dir=$(obj)/ $<; \
+	mv $(obj)/$(target-stem).d $(depfile); \
+	sed -i '/^\#/d' $(depfile)
```

Comment:
Does Make not treat those as comments?

Reply:
In a normal Make build system, yeah. However, Kbuild preprocess the dep files, so this is something we have in all calls to `rustc`.

Having said that, I think I will send a patch upstream to ignore comments and then we can remove this.

Reply:
:+1: Even better would be to actually interpret them the same way cargo does. Not sure of make supports rebuilding when an env var has changed from a specified value though. This is what these lines are used for.

Reply:
I think it could be implemented in Kbuild without too much trouble (since it is effectively doing its own dependency checking anyway).

But I think we could just avoid using `env!` instead. For instance, is there a way to extract the value of a `--cfg`? Because then we could simply reuse Kbuild's ability to track command-line changes (and would be useful for other purposes).

Reply:
> For instance, is there a way to extract the value of a --cfg?

There isn't AFAIK. Even if there was how would `--cfg foo="bar" --cfg foo="baz"` be handled? That is perfectly and neither cfg overrides the other. There is a PR open to add a commandline option to set env vars, but as I replied there, there is the problem that env vars are process global state and as such multiple rustc instances running in the same process would conflict. It may be possible to not update the process env var state and instead require `env!()` and proc macros to read a compilation session specific environment. (I believe proc macros already have an unstable api to read env vars to allow tracking env var accesses in the dep-info file.)

Reply:
It could error if there are several values defined in the "list" for that key, or it could take optionally an argument with the index, or simply take the last (which would mimic overriding mechanics) etc.; with a name like `cfg_value!(X)`.

I think we will need this at some point, even if it is not used for this, e.g. sometimes the kernel config stores integer values (not just booleans etc.). We can workaround it on our side (of course), but I think it is a useful feature for projects with lots of conditional compilation going on.

Reply:
The integer case would need some care, though, because we would like to have a way to be able to do something like:

```rust
let x = cfg_value!(X);
```

when `X` was an integer (i.e. no `"123"` but `123`). We could either to the str to int in a macro, but I guess the `--cfg` could be expanded non-quoted things too without breaking backwards compatibility?

Either that, or just an entire new feature that is not based on `--cfg` at all.

Reply:
I have uploaded https://github.com/ojeda/linux/commit/4cfe9f8eabacb1152b63790a13e5ce99c2fe21c1 as an example of how we could use it if `cfg_value!()` existed.

Do you think an RFC with some of those possible options is worth it? Would it be too disruptive on the compiler side?